javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/CloudFormationHelper.java

```diff
@@ -0,0 +1,157 @@
+                .build();
+        }
+        return cloudFormationClient;
+    }
+
+    public static void deployCloudFormationStack(String stackName) {
+        String templateBody;
+        boolean doesExist = describeStack(stackName);
+        if (!doesExist) {
+            try {
+                ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+                Path filePath = Paths.get(classLoader.getResource(CFN_TEMPLATE).toURI());
+                templateBody = Files.readString(filePath);
+            } catch (IOException | URISyntaxException e) {
+                throw new RuntimeException(e);
+            }
+
+            getCloudFormationClient().createStack(b -> b.stackName(stackName)
+                    .templateBody(templateBody)
+                    .capabilities(Capability.CAPABILITY_IAM))
+                .whenComplete((csr, t) -> {
+                    if (csr != null) {
+                        System.out.println("Stack creation requested, ARN is " + csr.stackId());
+                        try (CloudFormationAsyncWaiter waiter = getCloudFormationClient().waiter()) {
+                            waiter.waitUntilStackCreateComplete(request -> request.stackName(stackName))
+                                .whenComplete((dsr, th) -> {
+                                    if (th != null) {
+                                        System.out.println("Error waiting for stack creation: " + th.getMessage());
+                                    } else {
+                                        dsr.matched().response().orElseThrow(() -> new RuntimeException("Failed to deploy"));
+                                        System.out.println("Stack created successfully");
+                                    }
+                                }).join();
+                        }
+                    } else {
+                        System.out.format("Error creating stack: " + t.getMessage(), t);
+                        throw new RuntimeException(t.getCause().getMessage(), t);
+                    }
+                }).join();
+        } else {
+            logger.info("{} stack already exists", CFN_TEMPLATE);
+        }
+    }
+
+    // Check to see if the Stack exists before deploying it
+    public static Boolean describeStack(String stackName) {
+        try {
+            CompletableFuture<?> future = getCloudFormationClient().describeStacks();
+            DescribeStacksResponse stacksResponse = (DescribeStacksResponse) future.join();
+            List<Stack> stacks = stacksResponse.stacks();
+            for (Stack myStack :stacks) {
+                if (myStack.stackName().compareTo(stackName)==0){
+                    return true;
+                }
+            }
+        } catch (CloudFormationException e) {
+            System.err.println(e.getMessage());
+        }
+        return false;
+    }
+    public static void destroyCloudFormationStack(String stackName) {
```

Comment:
Need a line before this function.

Reply:
Fixed

Reply:
Did you miss a commit? It still looks wrong.